### PR TITLE
Fix filtering empty chat messages

### PR DESCRIPTION
**Pull request by Polygons1**
## What changed

- Preserve empty regular chat messages in `StatusBarTracker`
- Continue filtering empty action bar messages
- Add a regression test for empty non-overlay messages

## Why

`allowOverlayMessage` checked whether every game message was empty without first checking the `overlay` flag. As a result, empty regular chat packets were rejected by `ClientReceiveMessageEvents.ALLOW_GAME`. Hypixel uses those packets to add vertical spacing around multi-line messages such as Tree Gift rewards.

## Testing

- `./gradlew spotlessCheck checkstyleMain checkstyleTest test --tests de.hysky.skyblocker.skyblock.StatusBarTrackerTest`

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/StatusBarTracker.java [hunk 1]
```diff
@@ -120,8 +120,9 @@ private static InteractionResult interactItem(Player player, Level world, Intera
 		return InteractionResult.PASS;
 	}
 
-	private static boolean allowOverlayMessage(Component text, boolean overlay) {
-		return !onOverlayMessage(text, overlay).getString().isEmpty();
+	@VisibleForTesting
+	protected static boolean allowOverlayMessage(Component text, boolean overlay) {
+		return !overlay || !onOverlayMessage(text, true).getString().isEmpty();
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/StatusBarTracker.java]:
```suggestion
		return !overlay || !onOverlayMessage(text, overlay).getString().isEmpty();
```

**Review by viciscat — APPROVED** [review]

**Review by AzureAaron — APPROVED** [review]
